src/modals/contact-modals/edit-contact-modal/edit-contact-modal.tsx

```diff
@@ -13,30 +14,59 @@ import { Label } from '../../../components/label/label';
 import { ScreenContainer } from '../../../components/screen-container/screen-container';
 import { FormAddressInput } from '../../../form/form-address-input';
 import { FormTextInput } from '../../../form/form-text-input';
+import { useReadOnlyTezosToolkit } from '../../../hooks/use-read-only-tezos-toolkit.hook';
 import { AccountBaseInterface } from '../../../interfaces/account.interface';
 import { ModalsEnum, ModalsParamList } from '../../../navigator/enums/modals.enum';
 import { useNavigation } from '../../../navigator/hooks/use-navigation.hook';
 import { editContactAction, loadContactTezosBalance } from '../../../store/contact-book/contact-book-actions';
+import { useSelectedAccountSelector } from '../../../store/wallet/wallet-selectors';
 import { formatSize } from '../../../styles/format-size';
+import { showErrorToast } from '../../../toast/error-toast.utils';
+import { isTezosDomainNameValid, tezosDomainsResolver } from '../../../utils/dns.utils';
+import { isValidAddress } from '../../../utils/tezos.util';
 import { useEditContactFormValidationSchema } from '../validation-schema';
 import { EditContactModalSelectors } from './edit-contact-modal.selectors';
 
 export const EditContactModal: FC = () => {
+  const [isLoading, setIsLoading] = useState(false);
   const dispatch = useDispatch();
   const { goBack } = useNavigation();
   const {
     params: { contact, index }
   } = useRoute<RouteProp<ModalsParamList, ModalsEnum.EditContact>>();
   const editContactFormValidationSchema = useEditContactFormValidationSchema(index);
+  const selectedAccount = useSelectedAccountSelector();
+  const tezos = useReadOnlyTezosToolkit(selectedAccount);
+  const resolver = tezosDomainsResolver(tezos);
 
-  const onSubmit = (contact: AccountBaseInterface) => {
-    dispatch(editContactAction({ contact, index }));
-    dispatch(loadContactTezosBalance.submit(contact.publicKeyHash));
-    goBack();
+  const formik = useRef<FormikProps<typeof contact>>(null);
+
+  const onSubmit = async (contact: AccountBaseInterface) => {
```

Comment:
Too much code from https://github.com/madfish-solutions/templewallet-mobile/pull/806/commits/88b85993148e392f853cd9d5667c6fb696977b1b#diff-63a04e7fe55c6f563028b77e7fcf37e256f21faf0bc7d073ed3adefa56faca18R42 is here, please refactor